src/ServiceControl.AcceptanceTests/Recoverability/When_a_message_is_retried.cs

```diff
@@ -0,0 +1,115 @@
+﻿namespace ServiceBus.Management.AcceptanceTests.Recoverability
+{
+    using System;
+    using System.Collections.Generic;
+    using System.Threading.Tasks;
+    using EndpointTemplates;
+    using NServiceBus;
+    using NServiceBus.AcceptanceTesting;
+    using NServiceBus.MessageMutator;
+    using NServiceBus.Routing;
+    using NServiceBus.Transport;
+    using NUnit.Framework;
+    using ServiceControl.MessageFailures.Api;
+    using Conventions = NServiceBus.AcceptanceTesting.Customization.Conventions;
+
+    class When_a_message_is_retried : AcceptanceTest
+    {
+        static readonly List<string> HeadersThatShouldBeRemoved = new List<string>
```

Comment:
I know we try to avoid having internals visible to attribute. But I'm wondering if it would make sense to add one to ServiceControl to be able to reuse the headers that should be removed. Because now if we add an additional one that needs to be removed we might forget to add it here. What is your thoughts?

Reply:
On the fence, I do see the value here but opening up for accessing the internals might lead us to misuse in the future. Let's discuss that separately.